sage/example-transition/src/rules/hero_jam.rs

```diff
@@ -1,91 +1,119 @@
-use crate::{asset, asset::hero_jam, rules::RuleVerifier};
-
-use ajuna_primitives::{
-	asset_manager::{AssetInspector, AssetManager},
-	chain_inspector::ChainInspector,
+use crate::asset::{
+	hero_jam::{AssetType, HeroJamAsset, StateType},
+	Asset, AssetId,
 };
-use sage_api::Error;
 
-use crate::asset::AssetVariant;
-use frame_support::pallet_prelude::{Decode, Encode, MaxEncodedLen, TypeInfo};
-use parity_scale_codec::Codec;
-use sp_runtime::traits::{BlockNumber as BlockNumberT, Member};
-use std::marker::PhantomData;
+use ajuna_primitives::{asset_manager::AssetInspector, chain_inspector::ChainInspector};
+use sage_api::RuleError;
+use sp_runtime::traits::BlockNumber as BlockNumberT;
+
+pub const ASSET_NOT_FOUND: u8 = 200;
+pub const ASSET_NOT_HERO_JAM: u8 = 201;
+
+pub const ASSETS_NOT_ALL_SAME_TYPE: u8 = 100;
+pub const ASSETS_NOT_ALL_SAME_STATE: u8 = 101;
+pub const ASSET_CANNOT_STATE_CHANGE: u8 = 102;
+pub const ASSET_HERO_NOT_IN_ACCOUNT: u8 = 103;
+pub const ASSET_HERO_ALREADY_IN_ACCOUNT: u8 = 104;
+
+fn try_cast_to_hero_jam_asset<AccountId, BlockNumber, Inspector>(
```

Comment:
`cast` is a wrong term here. It is used with the `as` keyword. Into/From functions are called conversions.

Reply:
Actually, this is the wrong term anyhow, we just get the asset as hero jam. I would rename this into `try_get_hero_jam()`